programs/bench.h

```diff
@@ -19,46 +19,130 @@ extern "C" {
 #define ZSTD_STATIC_LINKING_ONLY   /* ZSTD_compressionParameters */
 #include "zstd.h"     /* ZSTD_compressionParameters */
 
+typedef enum {
+    BMK_timeMode = 0,
+    BMK_iterMode = 1
+} BMK_loopMode_t;
+
+typedef enum {
+    BMK_both = 0,
+    BMK_decodeOnly = 1,
+    BMK_compressOnly = 2
+} BMK_mode_t;
+
+#define ERROR_STRUCT(baseType, typeName) typedef struct { \
+    int error;       \
+    baseType result; \
+} typeName
+
 typedef struct {
     size_t cSize;
     double cSpeed;   /* bytes / sec */
     double dSpeed;
 } BMK_result_t;
 
-/* 0 = no Error */
 typedef struct {
-	int errorCode;
-	BMK_result_t result;
-} BMK_return_t;
+    size_t size;    /* sum of return values */
```

Comment:
Is the "sum of return values" always meaningful ?
In particular, what happens if one of the return values is an error code ?

Also : if this value is a sum of return values, better name it this way.
Here it's called `size`, but it might not be a size at all.